### DISCUSSION TITLE
Add: Option to hide feature (as cover) image

### DISCUSSION THREAD

**Discussion by ka1h** [jpanther/congo#418] [Ideas]
Hi everyone!

First of all: I'm a big fan of the theme, thank you so much for maintaining & contributing to it.

# What I'd like to change
Make the rendering of the `feature` image as cover image on pages optional via site parameter.

# Why I'd like to change it
Consider the following scenario: You might release a post/article/page accompanied with an adequate `feature.png`. So far, so good. But for stylistic reasons, you only want the `feature` image to appear as `og:meta` property for social embeds, but not actually rendered as cover image at the top of the corresponding page.

Take for example my CV page: https://ka1.io/cv/. I decided to not include a `feature` image, as i like the minimal top of the page. I end up with the following open graph link preview:
![Screenshot 2022-12-15 at 21 14 19](https://user-images.githubusercontent.com/51164387/207958096-d759977b-5274-4ee5-bbeb-f90c95d32700.png)

What I actually want to end up with:
![Screenshot 2022-12-15 at 21 14 27](https://user-images.githubusercontent.com/51164387/207958136-6b046a09-8797-4ef3-8b48-b1b0879e0272.png)

# How I'd like to change it
For example, in `_default/single.html` change the following block
```html
        {{ with $feature }}
            <img
            class="mb-6 -mt-4 rounded-md"
            srcset="
                    {{- (.Resize "330x").RelPermalink }} 330w,
                    {{- (.Resize "660x").RelPermalink }} 660w,
                    {{- (.Resize "1024x").RelPermalink }} 1024w,
                    {{- (.Resize "1320x").RelPermalink }} 2x"
            src="{{ (.Resize "660x").RelPermalink }}"
            />
        {{ end }}
```
to
```html
      {{ if .Params.showFeature | default (.Site.Params.article.showFeature | default true) }}
        {{ with $feature }}
            <img
            class="mb-6 -mt-4 rounded-md"
            srcset="
                    {{- (.Resize "330x").RelPermalink }} 330w,
                    {{- (.Resize "660x").RelPermalink }} 660w,
                    {{- (.Resize "1024x").RelPermalink }} 1024w,
                    {{- (.Resize "1320x").RelPermalink }} 2x"
            src="{{ (.Resize "660x").RelPermalink }}"
            />
        {{ end }}
      {{ end }}
```

Setting the site parameter to `default true` leaves the current standard intact, while giving me the flexibility to edit the page frontmatter to ignore the cover image -- for example my CV page:
```yaml
---
title: "Curriculum Vitae"
...
showFeature: false
---
```

# Conclusion
What I end up with is property `<meta property="og:image" content="http://localhost:1313/cv/feature.png">`, without the rendered cover image on the page. This gives me the flexibility to render high quality social open graph link previews without a possible unwanted cover image on the corresponding page.

Are there any easier solutions for this? What do you think about this? :)

Cheers,
Kai

**Comment by maijau** [comment]
What I am doing, without config but just standard functionality, is to include a separate preview image per post in the front matter, e.g.:

```
images:
- /posts/applemaps-cycling/og-preview.png
```

Note that due to an issue with the standard opengraph tag generation, you need to provide the path to the image.
Your configuration based solution is certainly nice, however with this simple approach you are able to use a dedicated picture for the link preview which could be different from other pics in the post.
